portality/bll/services/application.py

```diff
@@ -80,7 +80,7 @@ def reject_application(self, application, account, provenance=True, note=None, m
         if manual_update:
             application.set_last_manual_update()
 
-        saved = application.save()
+        saved = application.save(blocking=True)
```

Comment:
We try to avoid using `blocking` saves in the live application, as it can result in long hangs in the response times.  I suspect you've put this in here to aid a test, in which case you need to put the block or the `time.sleep` in the test instead.

Reply:
It is not for the test. It's the race caused somehow by the background job. I discussed that with @Steven-Eardley, still better than putting `time.sleep` on the background job.